src/app/api/v1/location-autocomplete/getRelevantLocations.ts

```diff
@@ -5,10 +5,26 @@
 import uFuzzy from "@leeoniya/ufuzzy";
 import { RelevantLocation } from "./types";
 
+// Guard against pathological inputs that would make ufuzzy's regex generation hang.
+export const MAX_SEARCH_QUERY_LENGTH = 128;
+
 export function getRelevantLocations(
   searchQuery: string,
   knownLocations: RelevantLocation[],
 ): RelevantLocation[] {
+  if (typeof searchQuery !== "string") {
+    return [];
+  }
+
+  const normalizedSearchQuery = searchQuery.trim();
+
+  if (
+    normalizedSearchQuery.length === 0 ||
+    normalizedSearchQuery.length > MAX_SEARCH_QUERY_LENGTH
```

Comment:
Suggestion: Should we log the query if someone exceeds the length? Could help us detect attack attempts, or valid searches that we are unintentionally blocking.